netbox_onboarding/netdev_keeper.py

```diff
@@ -96,7 +97,12 @@ def __init__(  # pylint: disable=R0913
         self.password = password
         self.secret = secret
         self.napalm_driver = napalm_driver
-        self.optional_args = optional_args
+
+        # Netmiko and NAPALM expects optional_args to be a dictionary.
+        if isinstance(optional_args, dict):
+            self.optional_args = optional_args
+        else:
+            self.optional_args = {}
```

Comment:
This will silently succeed if `optional_args` is not a `dict` (e.g. caller passed in a `list` by mistake). I'd prefer the one-liner:

```python
self.optional_args = optional_args or {}
```

Reply:
Only None or {} values are allowed, so instead of silently overriding lists etc, we could raise with following code :
```
        if isinstance(optional_args, dict):
            self.optional_args = optional_args
        elif optional_args is None:
            self.optional_args = {}
        else:
            raise OnboardException(reason="fail-general", message="Optional arguments should be None or a dict")
```